File: app/src/main/java/io/github/romangolovanov/apps/ametro/model/entities/MapPoint.java
package io.github.romangolovanov.apps.ametro.model.entities;

import io.github.romangolovanov.apps.ametro.render.utils.Algorithms;

public class MapPoint{
    public final float x;
    public final float y;

    public MapPoint(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public double distance(MapPoint point) {

        return Algorithms.calculateDistance(x,y,point.x, point.y);
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/model/entities/MapPoint.kt
package io.github.romangolovanov.apps.ametro.model.entities

import io.github.romangolovanov.apps.ametro.render.utils.calculateDistance

class MapPoint(val x: Float, val y: Float) {
    fun distance(point: MapPoint): Double {
        return calculateDistance(x, y, point.x, point.y).toDouble()
    }
}
